locomotion_manager/locomotion_manager.py

```diff
@@ -188,12 +192,13 @@ async def change_locomotion_mode_service_callback(self, request, response):
         self.get_logger().debug('Locomotion mode change request: %s' % request.new_locomotion_mode)
 
         future = self.state_machine.change_state(request.new_locomotion_mode)
-        result = await future
+        result, msg = await future
 
         if result is True:
             response.success = True
         else:
             response.success = False
+            response.message = msg  # "Can not change to the requested locomotion mode: " + request.new_locomotion_mode
```

Comment:
The message in the successful case is discarded here. Suggestion to change as follows. Also drop the comment.
```suggestion
        response.success = result
        response.message = msg
```

Reply:
Fixed